chainer/functions/activation/softmax.py

```diff
@@ -36,17 +37,17 @@ def forward(self, x):
             one = numpy.array(1, dtype=oz_dtype).ctypes
             zero = numpy.array(0, dtype=oz_dtype).ctypes
             handle = cudnn.get_handle()
-            x_cube = x[0].reshape(x[0].shape[:2] + (-1, 1))
+            x_cube = x[0].reshape(self._get_cube_shape(x[0].shape))
             desc = cudnn.create_tensor_descriptor(x_cube)
             self.y = xp.empty_like(x[0])
             libcudnn.softmaxForward(
                 handle, _algorithm, _mode, one.data, desc.value,
                 x_cube.data.ptr, zero.data, desc.value,
                 self.y.data.ptr)
         else:
```

Comment:
Ah, right. Sorry, this is my misunderstanding. I misunderstood the below error raised in Jenkins's test on Python2 env:

```
16:17:41 ======================================================================
16:17:41 ERROR: test_call_cudnn_backward (test_softmax.TestSoftmaxCudnnCall_param_10)  parameter: {'dtype': <type 'numpy.float64'>, 'use_cudnn': True, 'axis': 1}
16:17:41 ----------------------------------------------------------------------
16:17:41 Traceback (most recent call last):
16:17:41   File "/work/chainer/tests/chainer_tests/functions_tests/activation_tests/test_softmax.py", line 116, in test_call_cudnn_backward
16:17:41     y = self.forward()
16:17:41   File "/work/chainer/tests/chainer_tests/functions_tests/activation_tests/test_softmax.py", line 108, in forward
16:17:41     return functions.softmax(x, use_cudnn=self.use_cudnn, axis=self.axis)
16:17:41   File "/work/chainer/chainer/functions/activation/softmax.py", line 104, in softmax
16:17:41     return Softmax(use_cudnn=use_cudnn, axis=axis)(x)
16:17:41   File "/work/chainer/chainer/function.py", line 199, in __call__
16:17:41     outputs = self.forward(in_data)
16:17:41   File "/work/chainer/chainer/functions/activation/softmax.py", line 46, in forward
16:17:41     self.y.data.ptr)
16:17:41   File "cupy/cuda/cudnn.pyx", line 1060, in cupy.cuda.cudnn.softmaxForward (cupy/cuda/cudnn.cpp:12995)
16:17:41     cpdef softmaxForward(
16:17:41   File "cupy/cuda/cudnn.pyx", line 1068, in cupy.cuda.cudnn.softmaxForward (cupy/cuda/cudnn.cpp:12843)
16:17:41     check_status(status)
16:17:41   File "cupy/cuda/cudnn.pyx", line 417, in cupy.cuda.cudnn.check_status (cupy/cuda/cudnn.cpp:1835)
16:17:41     raise CuDNNError(status)
16:17:41 CuDNNError: CUDNN_STATUS_NOT_SUPPORTED: CUDNN_STATUS_NOT_SUPPORTED
```